massgen/system_message_builder.py

```
@@ -997,6 +1013,41 @@ def _get_tool_category_overrides(agent) -> dict[str, str]:
+        max_fix_loops = getattr(coord_cfg, "max_internal_fix_loops", None)
+        skip_scaffolding = getattr(coord_cfg, "skip_redundant_scaffolding", False)
+
+        if max_verifications is None and max_fix_loops is None and not skip_scaffolding:
+            return
+
+        scaffolding_exists = False
+        if skip_scaffolding and hasattr(agent, "backend") and hasattr(agent.backend, "filesystem_manager") and agent.backend.filesystem_manager:
+            try:
+                workspace_path = Path(agent.backend.filesystem_manager.get_current_workspace())
+                scaffolding_exists = (workspace_path / "tasks" / "changedoc.md").exists() or (workspace_path / "CONTEXT.md").exists() or (workspace_path / "tasks" / "plan.json").exists()
+            except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)